src/Schema.php

```diff
@@ -81,11 +81,7 @@ public function define(string $role, int $property)
             throw new SchemaException("Undefined schema `{$role}`, not found");
         }
 
-        if (!array_key_exists($property, $this->schema[$role])) {
-            return null;
-        }
-
-        return $this->schema[$role][$property];
+        return $this->schema[$role][$property] ?? null;
```

Comment:
same as before `$arr[$key]?? null` yields `null` if key does not exists or if the value is `null`, otherwise return the not-null value, but again faster than array_key_exists call. (pinginf @wolfy-j )

Reply:
...I forgot another matter:

getRelations() must return an array, but `define` may return `null`, and so it would array_keys on NULL inside getRelations. maybe I sould add another commit to check for null in getRelations and return an empty array in that case. @wolfy-j  kind regards

Reply:
> same as before $arr[$key]?? null yields null if key does not exists or if the value is null, otherwise return the not-null value, but again faster than array_key_exists call. (pinging @wolfy-j )

Confirmed.

> getRelations() must return an array, but define may return null, and so it would array_keys on NULL inside getRelations. maybe I sould add another commit to check for null in getRelations and return an empty array in that case. @wolfy-j kind regards

Yes please.